kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluatorSuite.scala

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.expressions
+
+import java.math.{BigDecimal => BigDecimalJ}
+
+import io.delta.kernel.data.{ColumnarBatch, ColumnVector}
+import io.delta.kernel.defaults.internal.data.DefaultColumnarBatch
+import io.delta.kernel.defaults.TestUtils
+import io.delta.kernel.defaults.internal.data.vector.VectorUtils.getValueAsObject
+import io.delta.kernel.expressions.{Column, Expression, Literal}
+import io.delta.kernel.expressions.predicates.AlwaysFalse.ALWAYS_FALSE
+import io.delta.kernel.expressions.predicates.AlwaysTrue.ALWAYS_TRUE
+import io.delta.kernel.types._
+import org.scalatest.funsuite.AnyFunSuite
+
+class DefaultExpressionEvaluatorSuite extends AnyFunSuite with TestUtils {
+  test("expression: literal") {
+    val testLiterals = Seq(
+      Literal.ofBoolean(true),
+      Literal.ofBoolean(false),
+      Literal.ofNull(BooleanType.INSTANCE),
+      Literal.ofByte(24.toByte),
+      Literal.ofNull(ByteType.INSTANCE),
+      Literal.ofShort(876.toShort),
+      Literal.ofNull(ShortType.INSTANCE),
+      Literal.ofInt(2342342),
+      Literal.ofNull(IntegerType.INSTANCE),
+      Literal.ofLong(234234223L),
+      Literal.ofNull(LongType.INSTANCE),
+      Literal.ofFloat(23423.4223f),
+      Literal.ofNull(FloatType.INSTANCE),
+      Literal.ofDouble(23423.422233d),
+      Literal.ofNull(DoubleType.INSTANCE),
+      Literal.ofString("string_val"),
+      Literal.ofNull(StringType.INSTANCE),
+      Literal.ofBinary("binary_val".getBytes),
+      Literal.ofNull(BinaryType.INSTANCE),
+      Literal.ofDate(4234),
+      Literal.ofNull(DateType.INSTANCE),
+      Literal.ofTimestamp(2342342342232L),
+      Literal.ofNull(TimestampType.INSTANCE))
+
+    val inputBatches: Seq[ColumnarBatch] = Seq[ColumnarBatch](
+      emptyColumnarBatch(0),
```

Comment:
It is very confusing to call this "empty". Empty collection means no rows, NOT no columns. so this callsite should be `zeroColumnBatch(rows = 0)`